### PR TITLE
Improve R2RDump method diff

### PR DESCRIPTION
**Pull request by trylek**
Based on my investigation of System.Private.CoreLib size growth
after switching over to Crossgen2 I have used R2RDump to analyze
the differences and this is an initial set of improvements I made
to make the diff more useful. Most notably it introduces
a statistic summary of method sizes and counts that were previously
lost in the lengthy tables of individual methods.

Thanks

Tomas

/cc @dotnet/crossgen-contrib

### CODE REVIEW

File: src/coreclr/src/tools/r2rdump/R2RDiff.cs [hunk 1]
```diff
@@ -75,46 +69,38 @@ public void Run()
...
+
+                default:
+                    methods.AddRange(reader.ReadyToRunAssemblies[moduleIndex].Methods);
+                    break;
             }
+            return methods;
         }
 
-        private void DiffMethodsForModule(ReadyToRunSection leftSection, ReadyToRunSection rightSection)
+        private void DiffMethodsForModule(int leftIndex, int rightIndex)
```
Comment (janvorli) [inline src/coreclr/src/tools/r2rdump/R2RDiff.cs:100]: A micro nit - would you mind naming the arguments `leftModuleIndex` and `rightModuleIndex` instead for better clarity of what kind of index they represent?

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

File: src/coreclr/src/tools/r2rdump/R2RDiff.cs (hunk 1 above)
Comment (trylek) [reply src/coreclr/src/tools/r2rdump/R2RDiff.cs]: Good point, renamed in 2nd commit (in three different places in the file).

**Review by janvorli — APPROVED** [review]

**Review by nattress — APPROVED** [review]
Looks great!

**Pull request merged**